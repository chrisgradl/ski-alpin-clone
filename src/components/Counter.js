import React, { useState } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

export const Counter = ({ initialValue = 0 }) => {
  const [count, setCount] = useState(initialValue);

  const increment = () => setCount((c) => c + 1);
  const decrement = () => setCount((c) => c - 1);

  return (
    <View>
      <View style={{ flexDirection: 'row' }}>
        <TouchableOpacity onPress={increment} accessibilityLabel="Increment">
          <Text>+</Text>
        </TouchableOpacity>
        <Text accessibilityLabel="counterValue">{count}</Text>
        <TouchableOpacity onPress={decrement} accessibilityLabel="Decrement">
          <Text>-</Text>
        </TouchableOpacity>
      </View>
      {count < 0 && (
        <Text accessibilityLabel="Info Message">Value is lower than 0</Text>
      )}
    </View>
  );
};
