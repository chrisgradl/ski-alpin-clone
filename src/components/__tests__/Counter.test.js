import { render, screen, fireEvent } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/react-native/extend-expect';

import { Counter } from '../Counter';

describe('Counter', () => {
  it('should have 0 as initial value and use prop as initial value if passed', () => {
    render(<Counter />);
    const counterText = screen.getByText('0');
    expect(counterText).toBeDefined();

    render(<Counter initialValue={5} />);
    const counterWithInitialValue = screen.getByText('5');
    expect(counterWithInitialValue).toBeDefined();
  });

  it('should count correctly', () => {
    render(<Counter />);
    const plusButton = screen.getByText('+');
    const minusButton = screen.getByText('-');
    fireEvent.press(plusButton);
    let textValue = screen.getByText('1');
    expect(textValue).toBeDefined();

    fireEvent.press(minusButton);
    textValue = screen.getByText('0');
    expect(textValue).toBeDefined();
  });

  it('should count correctly', () => {
    render(<Counter />);
    const minusButton = screen.getByText('-');
    fireEvent.press(minusButton);
    const infoText = screen.getByText('Value is lower than 0');
    expect(infoText).toBeDefined();
  });
});
