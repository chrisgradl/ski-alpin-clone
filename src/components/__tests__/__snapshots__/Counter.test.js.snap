// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly 1`] = `
<View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityLabel="Increment"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        +
      </Text>
    </View>
    <Text
      accessibilityLabel="counterValue"
    >
      0
    </Text>
    <View
      accessibilityLabel="Decrement"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        -
      </Text>
    </View>
  </View>
</View>
`;
